src/lumigo_tracer/parsers/event_parser.py

```diff
@@ -0,0 +1,75 @@
+import re
+from collections import OrderedDict
+from typing import Dict
+
+API_GW_REGEX = re.compile(rf".*execute-api.*amazonaws\.com.*")
+API_GW_KEYS_ORDER = [
+    "resource",
+    "path",
+    "httpMethod",
+    "queryStringParameters",
+    "multiValueQueryStringParameters",
+    "pathParameters",
+    "body",
+    "requestContext",
+    "headers",
+]
+
+API_GW_PREFIX_KEYS_HEADERS_DELETE_KEYS = (
+    "cookie",
+    "X-Amz",
+    "Accept",
+    "CloudFront",
+    "Via",
+    "X-Forwarded",
+    "sec-",
+)
+
+API_GW_REQUEST_CONTEXT_FILTER_KEYS = ["authorizer"]
+
+API_GW_KEYS_DELETE_KEYS = ["multiValueHeaders"]
+
+
+def parse_event(event: Dict):
+    try:
+        if _is_api_gw_event(event):
+            return _parse_api_gw_event(event)
+        else:
+            return event
+    except Exception:
+        return event
+
+
+def _is_api_gw_event(event: Dict) -> bool:
+    return API_GW_REGEX.match(event["requestContext"]["domainName"]) is not None
+
+
+def _parse_api_gw_event(event: Dict):
+    new_event: OrderedDict = OrderedDict()
+    # Add order keys
+    for order_key in API_GW_KEYS_ORDER:
```

Comment:
are you sure that there aren't any more keys?
We're going to omit all the other keys - maybe we want to keep them *first*?

Reply:
This is based on the AWS schema of API gateway.
All other keys are added in the end